executor/join.go

```diff
@@ -561,6 +561,17 @@ func (e *HashJoinExec) getNewJoinResult(workerID uint) (bool, *hashjoinWorkerRes
 
 func (e *HashJoinExec) join2Chunk(workerID uint, probeSideChk *chunk.Chunk, hCtx *hashContext, joinResult *hashjoinWorkerResult,
 	selected []bool) (ok bool, _ *hashjoinWorkerResult) {
+	killed := atomic.LoadUint32(&e.ctx.GetSessionVars().Killed) == 1
+	failpoint.Inject("killedInJoin2Chunk", func(val failpoint.Value) {
+		if val.(bool) {
+			killed = true
+		}
+	})
+	if killed {
+		joinResult.err = ErrQueryInterrupted
+		return false, joinResult
+	}
```

Comment:
Could this logic be moved into the for loop in runJoinWorker? So that there's no redundant code in `join2Chunk` and `join2ChunkForOuterHashJoin`?

Reply:
I think it can be moved into the for loop in runJoinWorker. Do you think so? @qw4990

Reply:
We can't. The purpose of adding these two checks is to let the join exit quickly when the inner table is large.
This issue is from an oncall-issue, in that case, the user processes a cartesian join with an inner table with 1e5 rows.
If we don't have this check, we have to process 1024(the size of batch) * 1e5 rows before exiting.

Reply:
After reviewing the PR again, I find that we should move these two checks into `for` loops in `join2Chunk`(L592) and `join2ChunkForOuterHashJoin `(L635).